### PR TITLE
Add logging in FileInStream and BlockInStream

### PR DESCRIPTION
**Pull request by apc999**
Add to `conf/log4j.properties`

```
log4j.logger.alluxio.client.block.stream.BlockInStream=DEBUG
log4j.logger.alluxio.client.file.AlluxioFileInStream=DEBUG
```

to enable client-side logs like:

```
2020-03-14 01:39:50,226 DEBUG AlluxioFuseFileSystem - Enter: read(path=/aaa,buf=jnr.ffi.provider.jffi.DirectMemoryIO[address=0x11e0d5000],size=65536,offset=0)
2020-03-14 01:39:50,227 DEBUG AlluxioFileInStream - Enter: read(file=/aaa,off=0,len=65536)
2020-03-14 01:39:50,261 DEBUG BlockInStream - Creating short circuit input stream for block 83886080 (/aaa) @ WorkerNetAddress{host=10.0.1.55, rpcPort=29999, dataPort=29999, webPort=30000, domainSocketPath=, tieredIdentity=TieredIdentity(node=10.0.1.55, rack=null)}
2020-03-14 01:39:50,314 DEBUG BlockInStream - Enter: seek(id=83886080,pos=0)
2020-03-14 01:39:50,314 DEBUG BlockInStream - Exit (OK): seek(id=83886080,pos=0)
2020-03-14 01:39:50,314 DEBUG BlockInStream - Enter: read(id=83886080,off=0,len=65536)
2020-03-14 01:39:50,324 DEBUG BlockInStream - Exit (6609): read(id=83886080,off=0,len=65536)
2020-03-14 01:39:50,324 DEBUG AlluxioFileInStream - Exit (6609): read(file=/aaa,off=0,len=65536)
2020-03-14 01:39:50,324 DEBUG AlluxioFuseFileSystem - Exit (6609): read(path=/aaa,buf=jnr.ffi.provider.jffi.DirectMemoryIO[address=0x11e0d5000],size=65536,offset=0) in 98 ms
```

One  can also set `alluxio.user.logging.threshold` e.g. to `10ms` to print Block / File level instream operations longer than `10ms`. Example output in logs:

```
2020-03-15 22:56:35,736 WARN  BlockInStream - seek(id=83886080,pos=0) returned null in 0 ms
2020-03-15 22:56:35,744 WARN  BlockInStream - read(id=83886080,b=[B@30d48072,off=0,len=65536) returned 6609 in 7 ms
```

### CODE REVIEW

File: core/client/fs/src/main/java/alluxio/client/LoggingUtils.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import alluxio.util.ConfigurationUtils;
+
+import org.slf4j.Logger;
+
+import java.io.IOException;
+
+/**
+ * Utils for client-side logging.
+ */
+public final  class LoggingUtils {
```
Comment (jiacheliu3) [inline core/client/fs/src/main/java/alluxio/client/LoggingUtils.java:25]: There's a `LogUtils` class already. Maybe merge these two?

File: core/client/fs/src/main/java/alluxio/client/LoggingUtils.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    } catch (Exception e) {
+      long duration = System.currentTimeMillis() - startMs;
+      logger.debug("Exit (ERROR): {}({}) in {} ms: {}, tid={}", methodName, debugDesc, duration,
+          e.toString(), tid);
+      if (duration >= THRESHOLD) {
+        logger.warn("{}({}) failed with exception [{}] in {} ms (>={}ms), tid={}", methodName,
+            String.format(description, args), e.toString(), duration, THRESHOLD, tid);
+      }
+      throw e;
+    }
```
Comment (jiacheliu3) [inline core/client/fs/src/main/java/alluxio/client/LoggingUtils.java:80]: calculate in finally{}?

File: core/client/fs/src/main/java/alluxio/client/LoggingUtils.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+   * @param <V> type of return value of the RPC call
+   * @param method the method to be executed
+   * @param logger the logger to use for this call
+   * @param methodName the human readable name of the RPC call
+   * @param description the format string of the description, used for logging
+   * @param args the arguments for the description
+   * @return the return value of the RPC call
+   * @throws IOException
+   */
+  public static <V> V callAndLog(Callable<V> method, Logger logger, String methodName,
```
Comment (jiacheliu3) [inline core/client/fs/src/main/java/alluxio/client/LoggingUtils.java:56]: Another alternative may be adding a LoggingCallable like below, which could be more flexible. 
```
  class LoggingCallable<T> implements Callable<T> {
    Callable<T> mDelegate;
    public LoggingCallable(Callable<T> callable) {
      mDelegate = callable;
    }

    @Override
    public T call() throws Exception {
      // log
      T ret = mDelegate.call();
      // calculate time and log
      return ret;
    }
  }
```

It's up to you

**Comment by apc999** [comment]
@jiacheliu3  thanks for reviewing this PR. This was originally an enhancement for some debugging session I worked together with @cheyang . Later on we found it is not the root cause.

@gpang do you feel something like this is helpful? It adds logging but also has more complexity in our core code. If you feel helpful, i could finish this PR.

**Pull request closed without merging**